server/src/main/java/io/crate/expression/reference/doc/lucene/SourceParser.java

```diff
@@ -326,6 +326,11 @@ private static Object parseByType(XContentParser parser, DataType<?> type) throw
                 ((BitStringType) elementType).length()
             );
             case NumericType.ID -> elementType.sanitizeValue(parser.text());
+            case UndefinedType.ID -> switch (parser.currentToken()) {
+                case VALUE_NUMBER -> parser.numberValue();
+                case VALUE_BOOLEAN -> parser.booleanValue();
```

Comment:
Is it possible to have `VALUE_NULL` here?

Reply:
No, the null case is handled in https://github.com/crate/crate/blob/c98034c605c49c108115c3e7e129d49fbf94f505/server/src/main/java/io/crate/expression/reference/doc/lucene/SourceParser.java#L295